sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client.py

```diff
@@ -844,3 +846,126 @@ def send_in_download_blob(request: PipelineRequest, **kwargs) -> MagicMock:
+                        "digest": "sha256:86fed9f0203a09f13cbbb9842132e9000eeff51b3de0d4ff66ee03ab0e860d1f",
+                        "architecture": "amd64",
+                        "os": "linux"
+                    },
+                    {
+                        "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+                        "imageSize": 566,
+                        "digest": "sha256:b808af65792ab617b9032c20fb12c455dc2bf5efe1af3f0ac81a129560772d35",
+                        "annotations": {
+                            "vnd.docker.reference.digest": "sha256:86fed9f0203a09f13cbbb9842132e9000eeff51b3de0d4ff66ee03ab0e860d1f",
+                            "vnd.docker.reference.type": "attestation-manifest"
+                        },
+                        "architecture": "unknown",
+                        "os": "unknown",
+                    }
+                ]
+            }
+            return json.dumps(manifest)
+        def send(request: PipelineRequest, **kwargs) -> MagicMock:
+            return MagicMock(
+                status_code=200,
+                content_type="application/json; charset=utf-8",
+                text=get_manifest,
+        )
+        
+        with ContainerRegistryClient(
+            endpoint=self.containerregistry_endpoint, transport = MagicMock(send=send)
+        ) as client:
+            manifests = client.list_manifest_properties(HELLO_WORLD)
+            for manifest in manifests:
+                assert manifest.architecture in ["amd64", "unknown"]
+                assert isinstance(manifest.architecture, str)
+                assert isinstance(manifest.architecture, ArtifactArchitecture)
+                assert manifest.architecture.name
+                assert manifest.architecture.value
+                assert manifest.operating_system in ["linux", "unknown"]
+                assert isinstance(manifest.operating_system, str)
+                assert isinstance(manifest.operating_system, ArtifactOperatingSystem)
+                assert manifest.operating_system.name
+                assert manifest.operating_system.value
+
+    def test_extended_enums(self):
+        with pytest.raises(AttributeError):
+            ArtifactArchitecture.foo
+        with pytest.raises(ValueError):
+            ArtifactArchitecture("foo")
+        foo = ArtifactArchitecture._extended("foo")
+        assert isinstance(foo, str)
+        assert isinstance(foo, ArtifactArchitecture)
+        assert not isinstance(foo, ArtifactOperatingSystem)
+        assert foo.name == "FOO"
+        assert foo.value == "foo"
+        assert foo == "foo"
+        assert str(foo) == "ArtifactArchitecture.FOO"
+        assert repr(foo) == "<ArtifactArchitecture.FOO: 'foo'>"
+
+        # Test None cases
+        assert ArtifactArchitecture._extended(None) is None
+        assert ArtifactArchitecture._extended("") is None
+        assert ArtifactArchitecture._extended("   ") is None
```

Comment:
I've chosen to interpret whitespace as None. Open to feedback there if we feel whitespace is worth preserving.